### DISCUSSION TITLE
Remove or greatly increase the default light limit

### DISCUSSION THREAD

**Discussion by BlackMoons** [FakeFishGames/Barotrauma#13982] [Ideas and suggestions]
Hi. I am wondering why the light limit even exists? I just tested my submarine (Tons of indicator lights everywhere) with unlimited lights vs 10, and the result was 3.8mS vs 3.71mS average draw time.

Why does this setting even exist? that is 2.5% difference in render time between extremes of the setting. And the default setting of 50 results in light flickering on many subs (even some vanilla subs), often coming up in the #baro-tech-support discord channel.

Can we just remove it? Or make it a 'shadow casting light' limit instead since those actually can affect performance? (And adjust its scroll range to say, 5~100/unlimited for shadow casting lights, say 25 by default?)

**Comment by fatglizzzy9** [comment]
please

**Comment by Regalis11** [comment]
I think the range and default value of this setting was configured way back when the performance of the lighting system was much worse, before we'd done a bunch of optimizations to the system. It does seem like they're unnecessarily low now.

Addressed in https://github.com/FakeFishGames/Barotrauma-development/pull/5657 (PR in our private repo):

- Changed the maximum value of the "light limit" setting from 200 to 500, changed the default from 50 to 100.
- The setting only affects shadow-casting lights which are usually what causes performance issues.

**Reply by BlackMoons** [reply]
Thank you very much for this change. It feels much more inline with what barotrauma's typical lighting engine performance is capable of on modern hardware.

**Reply by Regalis11** [reply]
The PR has now been merged to our private repo, coming up in the next unstable build.
